sycl/source/detail/graph_impl.cpp

```diff
@@ -627,44 +611,17 @@ bool graph_impl::clearQueues() {
   return AnyQueuesCleared;
 }
 
-void graph_impl::searchDepthFirst(
-    std::function<bool(std::shared_ptr<node_impl> &,
-                       std::deque<std::shared_ptr<node_impl>> &)>
-        NodeFunc) {
-  // Track nodes visited during the search which can be used by NodeFunc in
-  // depth first search queries. Currently unusued but is an
-  // integral part of depth first searches.
-  std::set<std::shared_ptr<node_impl>> VisitedNodes;
+bool graph_impl::checkForCycles() {
+  std::list<std::shared_ptr<node_impl>> MSchedule;
+  sortTopological(MRoots, MSchedule, false);
 
-  for (auto &Root : MRoots) {
-    std::deque<std::shared_ptr<node_impl>> NodeStack;
-    if (visitNodeDepthFirst(Root.lock(), VisitedNodes, NodeStack, NodeFunc)) {
-      break;
-    }
-  }
+  bool CycleFound = MSchedule.size() != MNodeStorage.size();
```

Comment:
A comment would be useful as to why the schedule not containing all the nodes means there's a cycle.